### PR TITLE
[11.0][IMP] document_page: menu, category search and wizard to create menus

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
* move menus to a higher level.
 * fix categories search view.
 * fix create menu action not being accessible in any way.
* Add activities to pages (mail.activity.mixin)

### CODE REVIEW

File: document_page/views/document_page.xml [hunk 1]
```diff
@@ -95,12 +95,12 @@
         <field name="name">document.page.menu.form</field>
         <field name="model">document.page</field>
         <field name="arch" type="xml">
-            <form string="Document Page">
+            <form string="Document Page" create="0">
```
Comment (pedrobaeza) [inline document_page/views/document_page.xml:98]: Why create=0?

Comment (LoisRForgeFlow) [reply document_page/views/document_page.xml:98]: This is the form view used from custom menuitems that you create through the wizard, IMO it doesn't make sense that you can create a new page from there, as it is supposed to be a wiki placed at convenience.

Comment (pedrobaeza) [reply document_page/views/document_page.xml:98]: OK, I don't have the context but I trust you

Comment (LoisRForgeFlow) [reply document_page/views/document_page.xml:98]: See this:
image

does it make sense to you a create button from a "static" wiki?

**Comment by LoisRForgeFlow** [comment]
Moving the main form view for pages from this:
image

to this:
image

IMO the focus must be in the content, the category and other fields are only filled once (on creation) and if you need to review them (revisions, history...) you just go to the bottom of the page.

**Review by AdriaGForgeFlow — CHANGES_REQUESTED** [review]
Tested on runbot:
I have created a new Page and Created a Menu:
image

If I, now, delete the page, the menu is still there and upon clicking on it the followin error appears:
image

**Comment by LoisRForgeFlow** [comment]
@ageficent That should be fixed in the last commit

**Review by AdriaGForgeFlow — APPROVED** [review]
LGTM :+1: Funtional test and code review

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
@pedrobaeza ready to merge?

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM code review

**Pull request merged**